### PR TITLE
fix: auth redirect working

### LINKED ISSUE #280: Error when go to the account page without user signed in
When we go to `/account` or `/account/admin` without user signed in we have this error:

Screenshot 2022-11-07 at 12 28 57 PM

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes
Currently when you try to access account pages but are not logged in you see a 400 error. These changes will ensure a user is redirected to the sign in page instead of seeing the 400 error.

co-authored @abdonrd 

Resolves #280

## Implementation details
Based on @abdonrd findings https://github.com/Qiskit/saiba/issues/280#issuecomment-1315913448

## Screenshots

https://user-images.githubusercontent.com/8097306/202035228-1ae99a4a-5455-49ce-9e12-09b8e9d3d8af.mov

### CODE REVIEW

**Comment by yoshiemuranaka** [comment]
thanks @abdonrd for finding and solving 🙏

**Review by abdonrd — DISMISSED** [review]
Thanks! 🎉

**Review by abdonrd — APPROVED** [review]

File: middleware/auth.ts [hunk 1]
```diff
@@ -12,9 +12,13 @@ export default defineNuxtRouteMiddleware(async (from) => {
...
 
-    const { data: refreshTokenData } = await refreshToken();
-    accessToken.value = refreshTokenData.access_token;
-    isTokenRefreshed.value = true;
+    try {
+      const { data: refreshTokenData } = await refreshToken();
+      accessToken.value = refreshTokenData.access_token;
+      isTokenRefreshed.value = true;
+    } catch {
+      userStore.reset();
```
Comment (Tansito) [inline middleware/auth.ts:20]: Taking a look to the PR I see that if we are not authenticated we are resetting the `userStore` two times. Maybe one of those are not necessary.

**Comment by Tansito** [comment]
Not 100% related with this issue/PR but with this change we made that if you are not logged and you want to enter in a course you are redirected to the signin page (it seems the unit page is using this middleware, I don't think it should).

I will open an issue to review and analyze the entire authentication workflow. Ideas are welcome.

**Comment by yoshiemuranaka** [comment]
Thanks for catching @Tansito! I also commented here https://github.com/Qiskit/saiba/pull/318#issuecomment-1317115028

it seems like when you go to a private course and refresh the page, the content does not load. I think maybe this could be fixed with #297 (store user creds in local storage)...what do you think?

**Comment by Tansito** [comment]
Yeah you're right @yoshiemuranaka , I think that once time we add the functionality from #297 we will be able to review the entire authentication flow. In fact, @abdonrd commented me too today that we should review it.

**Comment by abdonrd** [comment]
@yoshiemuranaka @Tansito I like the idea to use the same URL for all the courses (whether public or private): 

- `/course/:courseSlug`

And depending on the permissions of the current user, we let him see the content or not.

**Pull request merged**